generators/server/templates/build.gradle.ejs

```diff
@@ -376,6 +376,12 @@ dependencies {
     <%_ if (['mongodb', 'cassandra', 'couchbase'].includes(databaseType)) { _%>
     implementation "org.springframework.boot:spring-boot-starter-data-<%= databaseType %><% if (reactive) { %>-reactive<% } %>"
     <%_ } _%>
+    <%_ if (databaseType === 'neo4j') { _%>
+    implementation "org.neo4j.springframework.data:spring-data-neo4j-rx-spring-boot-starter:${spring_data_neo4j_rx_version}"
+    implementation "eu.michael-simons.neo4j:neo4j-migrations-spring-boot-starter:${neo4j_migrations_version}"
+    testImplementation "org.testcontainers:neo4j:1.12.4"
+    testImplementation 'org.testcontainers:junit-jupiter:1.12.4'
```

Comment:
Just a reminder for later, when the jhipster bom is released we have both the neo4j versions and the testcontainers bom available so we can remove explicit versions eventually here.